tests/test_methods.rs

```diff
@@ -405,6 +405,62 @@ fn method_with_lifetime() {
     );
 }
 
+#[pyclass]
+struct MethodWithPyClassArg {
+    #[pyo3(get)]
+    value: i64,
+}
+
+#[pymethods]
+impl MethodWithPyClassArg {
```

Comment:
The test should probably check `Option<&mut C>` and `PyRefMut<C>`, just for completeness.